app/forms/section3/LocationOfGoods.scala

```diff
@@ -69,23 +69,50 @@ object LocationOfGoods extends DeclarationPage {
     input.drop(3).toUpperCase.headOption.map(_.toString).exists(predicate)
   }
 
-  def form(implicit messages: Messages, codeListConnector: CodeListConnector): Form[LocationOfGoods] = Form(mapping)
+  def form(version: Int)(implicit messages: Messages, codeListConnector: CodeListConnector): Form[LocationOfGoods] =
+    version match {
+      case 3 | 5 => Form(mappingForVersion3And5)
+      case _     => Form(mappingForOtherVersions(version == 7))
+    }
 
+  val radioGroupId = "locationOfGoods"
+  val userChoice = "userChoice"
   val locationId = "glc"
 
-  private def mapping(implicit messages: Messages, codeListConnector: CodeListConnector): Mapping[LocationOfGoods] =
+  lazy val goodsLocations: Seq[String] = List(
+    "GBAUDEUDEUDEUGVM",
+    "GBAUDVRDOVDVRGVM",
+    "GBAUEUTEUTEUTGVM",
+    "GBAUFISFISFISGVM",
+    "GBAUHYMHEYHYMGVM",
+    "GBAUHLYHLDHLYGVM",
+    "GBAULIVLIVLIVGVM",
+    "GBAUPEDMILPEDGVM",
```

Comment:
Can we avoid hard coding these values here? This will be lost/forgotten if we ever need to update these in our reference data files.